src/main/java/javaslang/WrappedString.java

```diff
@@ -0,0 +1,2198 @@
+/*     / \____  _    ______   _____ / \____   ____  _____
+ *    /  \__  \/ \  / \__  \ /  __//  \__  \ /    \/ __  \   Javaslang
+ *  _/  // _\  \  \/  / _\  \\_  \/  // _\  \  /\  \__/  /   Copyright 2014-2015 Daniel Dietrich
+ * /___/ \_____/\____/\_____/____/\___\_____/_/  \_/____/    Licensed under the Apache License, Version 2.0
+ */
+package javaslang;
+
+import javaslang.collection.*;
+import javaslang.collection.HashMap;
+import javaslang.collection.Iterator;
+import javaslang.collection.Map;
+import javaslang.collection.Vector;
+import javaslang.control.None;
+import javaslang.control.Option;
+import javaslang.control.Some;
+
+import java.io.Serializable;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.Charset;
+import java.util.*;
+import java.util.HashSet;
+import java.util.function.*;
+import java.util.regex.PatternSyntaxException;
+import java.util.stream.Collector;
+
+/**
+ * TODO javadoc
+ */
+public final class WrappedString implements CharSequence, IndexedSeq<Character>, Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final WrappedString EMPTY = new WrappedString("");
+    
+    private final java.lang.String back;
+    
+    private WrappedString(java.lang.String javaString) {
```

Comment:
yes, that exactly what I thought - I had the same problem - but I was far less imaginative and called my `WrappedString` just String2 ... :)